src/fides/api/models/sql_models.py

```diff
@@ -191,6 +193,22 @@ def from_fideslang_obj(
             is_default=data_category.is_default,
         )
 
+    @staticmethod
+    def get_parent_categories_from_key(data_category_key: str) -> Set[str]:
+        """
+        Utility method to traverse "up" the taxonomy hierarchy and unpack
+        a given data category fides key into a set of fides keys that include its
+        parent fides keys.
+
+        Example inputs and outputs:
+            - `a.b.c` --> [`a.b.c`, `a.b`, `a`]
+            - `a` --> [`a`]
+        """
+        parent_categories = {data_category_key}
+        while data_category_key := data_category_key.rpartition(".")[0]:
+            parent_categories.add(data_category_key)
+        return parent_categories
+
```

Comment:
This looks familiar

EDIT: Ah yes! data uses. we could consider drying it up and sharing the method -

Reply:
I'll remove it for now, I didn't end up needing it